rpcs3/Emu/Audio/Cubeb/CubebBackend.cpp

```diff
@@ -396,9 +456,22 @@ long CubebBackend::data_cb(cubeb_stream* /* stream */, void* user_ptr, void cons
 	return nframes;
 }
 
-void CubebBackend::state_cb(cubeb_stream* /* stream */, void* user_ptr, cubeb_state state)
+void CubebBackend::state_cb(cubeb_stream* stream, void* user_ptr, cubeb_state state)
 {
+	if (!stream)
+	{
+		Cubeb.error("state_cb called with invalid stream");
+		return;
+	}
+
 	CubebBackend* const cubeb = static_cast<CubebBackend*>(user_ptr);
+	ensure(cubeb);
+
+	if (stream != cubeb->m_stream)
```

Comment:
Must be done under the lock

Reply:
I had to lock this under the m_state_cb_mutex (also see the change in Open() ).
There does not seem to be any stream mutex.

Reply:
Turns out the state_cb is called from cubeb while opening the stream.
I used a recursive lock for now. Better idea?